libtsuba/include/tsuba/WriteGroup.h

```diff
@@ -17,18 +17,32 @@ class WriteGroup {
   struct AsyncOp {
     std::future<katana::Result<void>> result;
     std::string location;
+    uint64_t accounted_size;
   };
 
   std::string tag_;
   std::list<AsyncOp> pending_ops_;
+  uint64_t outstanding_size_{0};
+  uint64_t errors_{0};
+  uint64_t total_{0};
+  katana::Result<void> last_error_{katana::ResultSuccess()};
 
   WriteGroup(std::string tag) : tag_(std::move(tag)){};
 
   /// Add future to the list of futures this descriptor will wait for, note
-  /// the file name for debugging
-  void AddOp(std::future<katana::Result<void>> future, std::string file);
+  /// the file name for debugging, if the operation is associated with a file
```

Comment:
".... the file name for debugging. If .... we are responsible for, note the size."